docs/tutorial/06-allow-snake-to-eat.md

```diff
@@ -0,0 +1,166 @@
+### 6. Allow snake to eat
+
+We have the food, but our snake can't eat it yet, and we don't want to starve our little friend.
+
+The goal of this chapter is to allow the snake to eat the food and grow, but let's first quickly recap how it works. The snake eats the food when it overlaps it with its head, as soon the snake eats it, it will grow of one unit (a new tile will be appended to its body) and a new food pellet will be placed in the snake world.
+
+In other words, we need to check if the snake's head has the same coordinates of the pellet after every movement. The function `move_snake/1` looks the natural place where to execute this check.
+
+```elixir
+defp move_snake(%{snake: snake} = state) do
+  %{body: body, size: size, direction: direction} = snake
+
+  # New head's position
+  [head | _] = body
+  new_head = move(state, head, direction)
+
+  # Place a new head on the tile that we want to move to
+  # and remove the last tile from the snake tail
+  new_body = List.delete_at([new_head | body], -1)
+
+  state
+  |> put_in([:objects, :snake, :body], new_body)
+  |> maybe_eat_food(new_head)
```

Comment:
👍  Should we drop the `maybe`? 

```elixir
state
|> put_in([:objects, :snake, :body], new_body)
|> eat_pellet(new_head)
```

What do. you think?

Reply:
For now not necessary in my opinion. I think the names are a bit off here - e.g. `move_snake` does more than moving the snake - but in general this is still fine I'd say.

The "problem" with `maybe_` that the post tasks about also does not really fit our case I'd say.

Reply:
> For now not necessary in my opinion. I think the names are a bit off here - e.g. move_snake does more than moving the snake - but in general this is still fine I'd say.

Good point, what about `advance_snake` ?

> The "problem" with `maybe_` that the post tasks about also does not really fit our case I'd say.

Sorry, but I didn't get your comment. Do you mean that the function name does not match the current use case since it does more stuff like placing the new pellet? 🤔 
I see the point, but unfortunately I cannot came with a better name.
I kinda like that the `grow_snake` and `place_pellet` functions are grouped together since they are related. What about `maybe_grow_snake` or `check_xxx` dunno 😞

Reply:
I guess whether it's "move" or "advance" does not really matter. As said, both is fine here in my opinion.

My point with the `maybe_` part was: the blog post has a more obvious scenario where one can get rid of the "maybe" part. Our case is different, we want **always** run this check.

Therefore: let's stick with the current naming - sorry for the confusion 🙈